### PR TITLE
Added Lock 11 / Auth0 9.0, Updated SSO, JWT Algorithm Upgrade Fixes

### PR DESCRIPTION
**Pull request by cocojoe**
**Lock**
Only lock that will be used is Lock 11.x (Currently Beta till stable release)
Upgrade Auth0.js to 9.x (Currently Beta till stable release)
 - Removed RememberLastLogin
 - SSO
   - Replaced `ssoData` with `checkSession`
   - Moved `checkSession` call to `wp-login` vs every page (heavy calls)
   - Handle implicit mode

**Admin Interface**
- Removed connections tab, link to client in Auth0 Dashboard added.
- Removed `use Lock 10`

**Wizard**
- Removed Lock preview in Step 2 Social
- Remove connections interface duplication, link to client in Auth0 Dashboard added.
- Auth0 Client Setup
  - Enable Cross Origin Authentication
  - Add URL Cross Origin Authentication Fallback
  - Add Web Origin
  - Remove Step 2 from Allowed Web Origin

**Database**
- Default Use Lock 10

**Upgrade**
- Force Default to Use Lock 10
- Upgrade CDN URLs Lock 11/Auth0 9
- Upgrade Auth0 Client
  - Enable Cross Origin Authentication
  - Add URL Cross Origin Authentication Fallback
  - Add Web Origin

**Docs**
- Removed remeberLastLogin notes

### CODE REVIEW

File: templates/auth0-sso-handler-lock10.php [hunk 1]
```diff
@@ -12,10 +12,14 @@
...
-    }
-  });
+  var options = <?php echo json_encode( $lock_options->get_sso_options() ); ?>;
+  webAuth.checkSession(options
+  , function (err, authResult) {
+      if (typeof(authResult) === 'undefined') {
+        return;
+      }
+      window.location = '<?php echo home_url( '/?auth0=1' ); ?>&code=' + authResult.code + '&state=' + authResult.state;
+    });
```
Comment (glena) [inline templates/auth0-sso-handler-lock10.php]: what if it is using the implicit flow?
in that case you might need to do the same than the login screen (post back to wordpress).

Or maybe we can use `response_mode=form_post` and in this case in particular make it go to the login callback and after that refresh the user agent.

File: templates/initial-setup/connections.php [hunk 2]
```diff
@@ -13,152 +13,39 @@
 
 			<h1><?php _e( "Configure your social connections", "wp-auth0" ); ?></h1>
 
-			<p class="a0-step-text"><?php _e( "If your WordPress site's visitors already have social network accounts, they can access your site with their existing credentials, or they can set up a username/password combination safeguarded by Auth0's password complexity policies and brute force protection.", "wp-auth0" ); ?></p>
+			<p class="a0-step-text"><?php _e( "If your WordPress site's visitors already have social network accounts, they can access your site with their existing credentials, or they can set up a username/password combination safeguarded by Auth0's password complexity policies and brute force protection. To configure your connections, you'll use the Auth0 Dashboard.", "wp-auth0" ); ?></p>
```
Comment (glena) [inline templates/initial-setup/connections.php]: 😍

File: templates/auth0-singlelogout-handler.php [hunk 3]
```diff
@@ -13,12 +13,17 @@
       domain:'<?php echo $domain; ?>'
     });
 
-    webAuth.client.getSSOData(function(err, data) {
-      if (!err && ( !data.sso || uuids != data.lastUsedUserID)) {
-        window.location = '<?php echo html_entity_decode( $logout_url ); ?>';
+    var options = <?php echo json_encode( $lock_options->get_sso_options() ); ?>;
+    webAuth.checkSession(options, function (err, authResult) {
+      console.log(err);
```
Comment (glena) [inline templates/auth0-singlelogout-handler.php]: remove console.log

**Comment by glena** [comment]
looks good

File: templates/auth0-singlelogout-handler.php (hunk 3 above)
Comment (cocojoe) [reply templates/auth0-singlelogout-handler.php]: oops 😄

File: lib/WP_Auth0_Routes.php [hunk 4]
```diff
@@ -44,10 +49,35 @@ public function custom_requests( $wp ) {
...
+			clientID: '$client_id',
+			domain: '$domain',
+			redirectUri: '$redirect_uri'
+		  });
+		  auth0.crossOriginAuthenticationCallback();
+		</script>
+		</head>
+		<body></body>
+		</html>	  
+EOT;
```
Comment (glena) [inline lib/WP_Auth0_Routes.php]: is this well closed? why dont use regular `"`?

Comment (cocojoe) [reply lib/WP_Auth0_Routes.php]: Really just style, for a single line I think " is best, for multiple lines I like heredoc.  http://php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc

**Pull request merged**